src/main/scala/pl/agh/harmonytools/model/ChordComponent.scala

```diff
@@ -0,0 +1,41 @@
+package pl.agh.harmonytools.model
+
+import scala.collection.immutable.HashMap
+
+case class ChordComponent(
+  chordComponentString: String,
+  baseComponent: Int,
+  semitonesNumber: Int,
+  isDown: Boolean
+) {
+  lazy val toXmlString: String =
+    chordComponentString.replace("<", "&lt;").replace(">", "&gt;")
+}
+
+object ChordComponent {
+
+  final val baseComponentPitch: HashMap[Int, Int] = HashMap(
+    1 -> 0,
+    2 -> 2,
+    3 -> 4,
+    4 -> 5,
+    5 -> 7,
+    6 -> 9,
+    7 -> 10,
+    8 -> 12,
+    9 -> 14
+  )
+
+  def apply(chordComponentString: String, isDown: Boolean = false): ChordComponent = {
```

Comment:
Ciągle się zastanawiam czy powinniśmy przeliczać te wszystkie rzeczy (nie tylko tu... o wszystkich konstruktorach mówię) na froncie, podczas mapowania z obiektów restowych czy też tu w modelu... i chyba postawiłbym że należy to zrobić przy mapowaniu z resta, zarazem tam zapewniając poprawność tych obiektów